ion/textutils.go

```diff
@@ -9,6 +9,24 @@ import (
 	"time"
 )
 
+const (
+	// LayoutYYYYMMDDTHHMMZHHMM time.date format. time.Parse() uses
+	// "2006-01-02T15:04Z07:00" explicitly as a string value to identify this format.
+	LayoutYYYYMMDDTHHMMZHHMM = "2006-01-02T15:04Z07:00"
```

Comment:
We cannot use ```_``` in constant names (linter).
I am open to suggestions if anyone can think of better names.

Reply:
> We cannot use _ in constant names (linter).

Welp, it only took us a few hours to stub our collective toe on the linter. 😛 

At a glance, it's pretty hard to tell the difference between the names. `LayoutYYYYMMDDTHHMMZHHMM` and `LayoutYYYYMMDDTHHMMZ` are visually very similar. Maybe there's a good name set that emphasizes what's different about each option? (i.e. the resolution/suffixes)

How about:
```go
LayoutMinutesZulu // ('Z')
LayoutMinutesAndOffset // ('ZHHMM')
LayoutSecondsAndOffset  // RFC3339
LayoutNanosecondsAndOffset // RFC3339Nano
```
?

Reply:
You presumably want them lower-case so they're not part of the public API?

My vote would be to use the `time.` constants directly, and add a local definition for `const rfc3339Minute = "2006-01-02T15:04Z07:00"`.

Reply:
Thanks David! That was an oversight on my part - still sometimes I forget Capital letter means Public!
However, we might keep ```LayoutMinutesAndOffset``` naming as it is more descriptive than ```RFC3339``` for a person who is as clueless as me on what RFC3339 means at first glance.